### PR TITLE
chore: make webhooks ingress part of webhooks section in chart

### PR DESCRIPTION
**Pull request by eddycharly**
This PR makes webhooks ingress part of the webhooks section in chart.

This improves readability and allows the end user to specify the host he wants (works better with ngrok for example).

### CODE REVIEW

File: charts/lighthouse/templates/webhooks-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.webhooks.ingress.enabled }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "webhooks.name" . }}
```
Comment (abayer) [inline charts/lighthouse/templates/webhooks-ingress.yaml:5]: For legacy and Prow compatibility reasons, the ingress name still needs to be `hook`. That's why https://github.com/jenkins-x/lighthouse/blob/master/test/e2e/helpers.go#L347 failed in the e2e tests.

File: charts/lighthouse/values.yaml [hunk 2]
```diff
@@ -92,7 +92,6 @@ webhooks:
 
   # webhooks.service -- Service settings for the webhooks controller
   service:
-    name: hook
```
Comment (abayer) [inline charts/lighthouse/values.yaml:95]: Still need to keep this for compat reasons. =)

Comment (eddycharly) [reply charts/lighthouse/values.yaml:95]: I’ll add an override mechanism and state in the docs that it’s there for compact reasons (a regular user should not need it).

Comment (eddycharly) [reply charts/lighthouse/values.yaml:95]: Both the name of the service and ingress matter ?

Comment (abayer) [reply charts/lighthouse/values.yaml:95]: Regrettably, yes - Jenkins X makes some hard assumptions on both.

Comment (eddycharly) [reply charts/lighthouse/values.yaml:95]: Ok, i adapted to chart templates but i think i'll need to change some config for the tests too.

Comment (abayer) [reply charts/lighthouse/values.yaml:95]: Actually, the ingress name can be changed - I forgot that jx has its own hook ingress. But yeah, the Tekton tests will need to change if the ingress name changes.

Comment (eddycharly) [reply charts/lighthouse/values.yaml:95]: We could look up the ingress by labels instead of relying on an hard coded name.

**Comment by abayer** [comment]
The Jenkins X contexts are failing because they expect the service to be named hook.

**Comment by abayer** [comment]
It's annoying but hard-coded in https://github.com/jenkins-x-charts/jxboot-resources/blob/master/jxboot-resources/templates/700-hook-ing.yaml#L20

**Comment by eddycharly** [comment]
It should be named `hook` (i added `.Values.webhooks.serviceName`)

**Comment by eddycharly** [comment]
Ah no, there's no `values.yaml` files in those tests...

**Comment by abayer** [comment]
And we can't assume there would be, so at least for now, it has to default to `hook`.

**Comment by eddycharly** [comment]
Actually it looks like there is one in `bdd/values.yaml.template`.
But some keys don't look correct (tide ?).

**Comment by abayer** [comment]
That is rather old, but more importantly, it's meant to match up with https://github.com/jenkins-x/jenkins-x-boot-config/blob/master/env/lighthouse/values.tmpl.yaml as closely as possible, with the only changes that really matter being the `$VERSION` stuff. A Jenkins X install is going to make assumptions about service names, etc, so we really do need to keep it as `hook` for the foreseeable future.

**Comment by eddycharly** [comment]
Alright, i've set the default to be `hook` and changed the doc:
```
  # webhooks.serviceName -- Allows overriding the service name, this is here for compatibility reasons, regular users should clear this out
  serviceName: hook
```

**Comment by abayer** [comment]
Someday, once the Jenkins X release process is unborked (things are just..busted there at the moment), I'll update it to set `webhooks.serviceName` to `hook` so we can get rid of the explicit naming here.

**Pull request merged**